File: L02_3/app/src/main/java/com/example/l02_3/SecondActivity.java
package com.example.l02_3;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class SecondActivity extends AppCompatActivity {

    int request_Code = 1;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        String res = getIntent().getStringExtra("key");
        EditText actText = (EditText)findViewById(R.id.txtUsername);
        actText.setText(res);
        Toast.makeText(this, res, Toast.LENGTH_SHORT).show();
    }

    public void onClick(View view) {
        Intent data = new Intent();
        EditText txt_username = (EditText)findViewById(R.id.txtUsername);
        data.setData(Uri.parse(txt_username.getText().toString()));
        setResult(RESULT_OK, data);
        finish();
    }
}

File: L04_2/app/src/main/java/com/example/l04_2/SecondActivity.kt
package com.example.l04_2

import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity


class SecondActivity : AppCompatActivity() {

    private val requestCode = 10;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        val res = intent.getStringExtra("kotlin_key")
        val textBox = findViewById<EditText>(R.id.txtBox_second)
        textBox.setText(res)
        Toast.makeText(this, res, Toast.LENGTH_SHORT).show()
    }

    fun onClick(view: View) {
        val data = Intent()
        val textBox: EditText = findViewById(R.id.txtBox_second)
        data!!.data = Uri.parse(textBox.text.toString())
        setResult(Activity.RESULT_OK, data)
        finish()
    }
}
